dotcom-rendering/src/layouts/ShowcaseLayout.tsx

```diff
@@ -202,13 +204,23 @@ const PositionHeadline = ({
 	}
 };
 
-interface Props {
+interface CommonProps {
 	article: DCRArticle;
-	NAV: NavType;
 	format: ArticleFormat;
+	renderingTarget: RenderingTarget;
+}
+
+interface WebProps extends CommonProps {
+	NAV: NavType;
+	renderingTarget: 'Web';
```

Comment:
should we be passing the rendering target as a prop here? my understanding was we'd be using context to retrieve this value now

Reply:
Using contexts helps us avoid prop drilling in many cases, but in the cases where we want a component to have different props depending on whether it's rendering for web or apps, we still need to pass a `renderingTarget` prop. That's so we can leverage typescript to say that `NAV` only exists on `props` when `renderingTarget === 'Web'`. Otherwise, we would need the `NAV` prop on apps renders, where it doesn't exist and isn't used. We do a similar thing for `StandardLayout`:

https://github.com/guardian/dotcom-rendering/blob/e93779ac646041ae8227cf0fd128414c122c52f6/dotcom-rendering/src/layouts/StandardLayout.tsx#L290-L303

Because of the way Typescript/contexts work, the context pattern doesn't quite solve all of our use-cases like this one, and so we still need to use both approaches. I appreciate that this is a bit confusing, and it's one of the drawbacks (imo) of the contexts pattern in DCR.

Reply:
Ahh makes total sense! Thanks for the detailed writeup, George!